src/vivarium_gates_nutrition_optimization_child/components/population.py

```diff
@@ -77,6 +81,7 @@ def setup(self, builder: Builder) -> None:
 
         self.start_time = get_time_stamp(builder.configuration.time.start)
         self.location = self._get_location(builder)
+        builder.time.register_step_modifier(self.modify_step)
```

Comment:
Out of scope for now, but I wonder if we should make `modify_step` a "special" function name for `Component` objects like `on_time_step`, etc. By this I mean have component's `setup_component` function look for this method and automatically call `builder.time.register_step_modifier` so the component implementer doesn't need to remember to do it (and remove boilerplate).